js/plugins/checks/src/middleware.ts

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ModelMiddleware } from '@genkit-ai/ai/model';
+import { GoogleAuth } from 'google-auth-library';
+import { Guardrails } from './guardrails';
+import { ChecksEvaluationMetric } from './metrics';
+
+export function checksMiddleware(options: {
+  auth: GoogleAuth;
+  metrics: ChecksEvaluationMetric[];
+  projectId?: string;
+}): ModelMiddleware {
+  const guardrails = new Guardrails(options.auth, options?.projectId);
+
+  const classifyContent = async (content: string) => {
+    const response = await guardrails.classifyContent(content, options.metrics);
+
+    // Filter for violations
+    const violatedPolicies = response.policyResults.filter(
+      (policy) => policy.violationResult === 'VIOLATIVE'
+    );
+
+    return violatedPolicies;
+  };
+
+  return async (req, next) => {
+    for (const message of req.messages) {
+      for (const content of message.content) {
+        if (content.text) {
+          const violatedPolicies = await classifyContent(content.text);
+
+          // If any input message violates a checks policy. Stop processing,
+          // return a blocked response and list of violated policies.
+          if (violatedPolicies.length > 0) {
+            return {
+              finishReason: 'blocked',
+              finishMessage: `Model input violated Chec: [${violatedPolicies.map((result) => result.policyType).join(' ')}], further processing blocked.`,
+            };
```

Comment:
@pavelgj In the case of streaming. When a user passes in policy violating content to a streaming flow. This case is triggered. 

In the UI it seems to hang forever. The loading spinner next to the run button just keeps spinning. 

This is not the case when one of the messages streamed output violates policy. That works fine.

Reply:
there's a bug in in error handling, fixing it